meveo-admin-ejbs/src/main/java/org/meveo/service/git/GitClient.java

```diff
@@ -412,17 +414,36 @@ public void push(GitRepository gitRepository, String username, String password)
         }
 
         final File repositoryDir = GitHelper.getRepositoryDir(user, gitRepository);
-
+        String pushErrorMessage = "Updates were rejected because the remote contains work that you do not have locally.";
         keyLock.lock(gitRepository.getCode());
 
         try (Git git = Git.open(repositoryDir)) {
             final PushCommand push = git.push();
             if(gitRepository.getRemoteOrigin().startsWith("http")) {
-                CredentialsProvider usernamePasswordCredentialsProvider = GitHelper.getCredentialsProvider(gitRepository, username, password, user);
-                push.setCredentialsProvider(usernamePasswordCredentialsProvider).call();
+		        CredentialsProvider usernamePasswordCredentialsProvider = GitHelper.getCredentialsProvider(gitRepository, username, password, user);
+		        Iterable<PushResult> pushResults = push.setCredentialsProvider(usernamePasswordCredentialsProvider).call();
+		        for (PushResult pushresult : pushResults) {
+		            for(RemoteRefUpdate remoteRef : pushresult.getRemoteUpdates()) {
+		                if(remoteRef.getStatus() != RemoteRefUpdate.Status.OK && remoteRef.getStatus() != RemoteRefUpdate.Status.UP_TO_DATE) {
+		                    if(remoteRef.getStatus() == RemoteRefUpdate.Status.REJECTED_NONFASTFORWARD) {
```

Comment:
you should throw an exception in any case the status is not OK or UP_TO_DATE, not only in the case it is REJECTED_NONFASTFORWARD